include/onnxruntime/core/common/gsl.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "gsl/gsl"
+
+// hack in a version of gsl::narrow for builds with no exceptions
+// adapted from "gsl/narrow"
```

Comment:
MS GSL's gsl::narrow is unavailable in a build without exceptions since it unconditionally uses exceptions.
https://github.com/microsoft/GSL/blob/10df83d292bf5bbdc487e57dc8c2dc8c7a01f4d1/include/gsl/narrow#L57
we could potentially wrap it in onnxruntime::narrow to get a consistent API across builds.